runtime/amd64.S

```diff
@@ -734,16 +739,32 @@ CFI_STARTPROC
         C stack args        : begin=%r13 end=%r12 */
     /* Switch from OCaml to C */
         SWITCH_OCAML_TO_C
-    /* we use %rbx (otherwise unused) to enable backtraces */
-        movq    %rsp, %rbx
-#ifdef ASM_CFI_SUPPORTED
-        .cfi_escape DW_CFA_def_cfa_expression, 5,           \
-          /* %rbp points to the c_stack_link structure */   \
-          DW_OP_breg + DW_REG_rbx, Cstack_sp, DW_OP_deref,  \
-          DW_OP_plus_uconst, RETADDR_ENTRY_SIZE
-#endif
     /* Make the alloc ptr available to the C code */
         movq    %r15, Caml_state(young_ptr)
+    /* Copy the arguments and call */
+        C_call  (GCALL(caml_c_call_copy_stack_args))
+    /* Prepare for return to OCaml */
+        movq    Caml_state(young_ptr), %r15
+    /* Load ocaml stack and restore global variables */
+        SWITCH_C_TO_OCAML
+    /* Return to OCaml caller */
+        LEAVE_FUNCTION
+        RET_FROM_C_CALL
+CFI_ENDPROC
+ENDFUNCTION(G(caml_c_call_stack_args))
+
+/* To correctly maintain frame pointers during stack reallocation,
+   the runtime assumes that the caml_c_call stub does not push
+   anything to the stack before the first frame pointer on the C stack.
+   To guarantee this when stack arguments are used, the actual pushing
+   of arguments is done by this separate function */
+FUNCTION(G(caml_c_call_copy_stack_args))
+CFI_STARTPROC
+    /* Set up a frame pointer even without WITH_FRAME_POINTERS,
+       which we use to pop an unknown number of arguments later */
+        pushq   %rbp; CFI_ADJUST(8)
```

Comment:
Is there a reason why you don't use `ENTER_FUNCTION` and `LEAVE_FUNCTION`, like arm64, in this function?